pychromecast/socket_client.py

```diff
@@ -268,13 +268,16 @@ def initialize_connection(self):
                 self._report_connection_status(
                     ConnectionStatus(CONNECTION_STATUS_FAILED,
                                      NetworkAddress(self.host, self.port)))
-                retry_log_fun("Failed to connect, retrying in %.1fs",
-                              self.retry_wait)
-                retry_log_fun = self.logger.debug
 
-                time.sleep(self.retry_wait)
-                if tries:
+                # Only sleep if we have another retry remaining
+                if tries and tries > 1:
```

Comment:
If tries is 1, we will now never get to 0.

Reply:
Hah, duh, sorry.